Source/ToolCore/NETTools/NETProjectSystem.cpp

```diff
@@ -421,6 +425,96 @@ namespace ToolCore
 
     }
 
+    /// handle the results of a recoompile, and auto play the project if successful
+    void NETProjectSystem::HandleNETBuildPlay(StringHash eventType, VariantMap& eventData)
+    {
+        using namespace NETBuildResult;
+        
+        if (eventData[P_SUCCESS].GetBool())
+        {
+            ATOMIC_LOGINFOF("NETBuild Success for project, now playing.");
+            VariantMap shortcutData;  // encode a shortcut even to send a play
+            shortcutData[UIShortcut::P_KEY] = KEY_P;
+            shortcutData[UIShortcut::P_QUALIFIERS] = QUAL_CTRL;
+            SendEvent(E_UISHORTCUT, shortcutData);
+        }
+        else
+        {
+            const String& errorText = eventData[P_ERRORTEXT].GetString();
+
+            ATOMIC_LOGERRORF("\n%s\n", errorText.CString());
+            ATOMIC_LOGERRORF("NETBuild Error for project, will not play.");
+        }
+
+    }
+
+    /// used to ensure the project can be run from the editor
+    bool NETProjectSystem::BuildAndRun()
+    {
+        bool shouldRebuild = CheckForRebuild();
+        if (shouldRebuild) 
+        {
+            FileSystem* fileSystem = GetSubsystem<FileSystem>();
+            if (!fileSystem->FileExists(solutionPath_))
+            {
+                if (!GenerateSolution())
+                {
+                    ATOMIC_LOGERRORF("NETProjectSystem::BuildAtomicProject - solutionPath does not exist: %s", solutionPath_.CString());
+                    return false;
+                }
+            }
+
+            Project* project = GetSubsystem<ToolSystem>()->GetProject();
+            NETBuildSystem* buildSystem = GetSubsystem<NETBuildSystem>();
+            if (buildSystem)
+            {
+                NETBuild* build = buildSystem->BuildAtomicProject(project);
+                if (build)
+                {
+                    build->UnsubscribeFromEvent(E_NETBUILDRESULT); //unsubscribe in case it has been subscribed before
+                    build->SubscribeToEvent(E_NETBUILDRESULT, ATOMIC_HANDLER(NETProjectSystem, HandleNETBuildPlay));
+                }
+            }       
+        }
+        
+        return shouldRebuild;
+    }
+    
+    /// interrogate sources to see if any are dirty compared to the editor project.dll
+    bool NETProjectSystem::CheckForRebuild() 
+    {
```

Comment:
It does depend on the existing project structure, you need to be able to trust the assetdatabase. There is always room for improvement.